Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1354,19 +1368,72 @@ static bool isContextMenuEvent(const WebMouseEvent& event)
 void UnifiedPDFPlugin::scrollToPDFDestination(PDFDestination *destination)
 {
     auto unspecifiedValue = get_PDFKit_kPDFDestinationUnspecifiedValue();
+
+    auto pageIndex = [m_pdfDocument indexForPage:[destination page]];
     auto pointInPDFPageSpace = [destination point];
     if (pointInPDFPageSpace.x == unspecifiedValue)
         pointInPDFPageSpace.x = 0;
     if (pointInPDFPageSpace.y == unspecifiedValue)
-        pointInPDFPageSpace.y = 0;
+        pointInPDFPageSpace.y = pageHeight(pageIndex);
 
-    scrollToPointInPDF(roundedIntPoint(pointInPDFPageSpace), [m_pdfDocument indexForPage:[destination page]] - 1);
+    scrollToPointInPage(roundedIntPoint(pointInPDFPageSpace), pageIndex);
 }
 
-void UnifiedPDFPlugin::scrollToPointInPDF(IntPoint pointInPDFPageSpace, PDFDocumentLayout::PageIndex pageIndex)
+void UnifiedPDFPlugin::scrollToPointInPage(IntPoint pointInPDFPageSpace, PDFDocumentLayout::PageIndex pageIndex)
 {
-    auto pointInDocumentSpace = convertFromPageToContents(pointInPDFPageSpace, pageIndex);
-    scrollToPositionWithoutAnimation(pointInDocumentSpace);
+    auto pointInContentsSpace = convertFromPageToContents(pointInPDFPageSpace, pageIndex);
+    scrollToPositionWithoutAnimation(pointInContentsSpace);
+}
+
+void UnifiedPDFPlugin::scrollToPage(PDFDocumentLayout::PageIndex pageIndex)
+{
+    scrollToPointInPage({ 0, static_cast<int>(pageHeight(pageIndex)) }, pageIndex);
```

Comment:
Is the `pageHeight()` because of flipped coordinates in the page? Is a bit mysterious; a comment or some `topOfPage()` helper  would clarify.